doc/book/cookbook/flash-messengers.md

```diff
@@ -37,19 +37,32 @@ Second, create middleware that will add the flash message provider to the reques
 <?php
 namespace App;
 
-use Interop\Http\ServerMiddleware\DelegateInterface;
+// Expressive 3.X:
+use Interop\Http\Server\MiddlewareInterface;
+use Interop\Http\Server\RequestHandlerInterface;
+
+// Expressive 2.X:
+use Interop\Http\ServerMiddleware\DelegateInterface as RequestHandlerInterface;
 use Interop\Http\ServerMiddleware\MiddlewareInterface;
+
+use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface;
 use Slim\Flash\Messages;
 
 class SlimFlashMiddleware implements MiddlewareInterface
 {
-    public function process(ServerRequestInterface $request, DelegateInterface $delegate)
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler) : ResponseInterface
```

Comment:
Is the space before colon part of ZF coding standard? The PSR-12 draft recommends using no space before the colon.

Reply:
Yes. And PSR-12 is not doing a hard-recommendation on it, only a soft one.